src/Particle/HDFWalkerInput_0_4.h

```diff
@@ -81,11 +82,11 @@ struct HDFWalkerInput_0_4
   void checkOptions(xmlNodePtr cur);
 
   /** read walkers for small number of MPI tasks */
-  bool read_hdf5(std::string h5name);
+  bool read_hdf5(std::filesystem::path h5name);
   /** read walkers. Master reads and scatter the walkers */
-  bool read_hdf5_scatter(std::string h5name);
+  bool read_hdf5_scatter(std::filesystem::path h5name);
   /** read walkers using PHDF5 */
-  bool read_phdf5(std::string h5name);
+  bool read_phdf5(std::filesystem::path h5name);
```

Comment:
For the above three APIs, can we have `const std::filesystem::path&`?

Reply:
These functions all call `h5name.concat(hdf::config_ext);` which modifies the value of h5name.

Reply:
That is the confusing part. I feel the argument should be renamed as h5name_noext and kept intact and then h5name is used for the full name.